winter-framework/src/main/java/de/uni_mannheim/informatik/dws/winter/matching/rules/LinearCombinationMatchingRule.java

```diff
@@ -95,8 +95,9 @@ public LinearCombinationMatchingRule(double offset, double finalThreshold) {
 	public void addComparator(Comparator<RecordType, SchemaElementType> comparator, double weight) throws Exception {
 		if (weight > 0.0) {
 			comparators.add(new Pair<Comparator<RecordType, SchemaElementType>, Double>(comparator, weight));
-			for(int i = 0; i < this.headerResultsComparator.length; i++){
-				this.addColumnToResults(this.headerResultsComparator[i] + comparators.size());
+			if(this.isCollectDebugResults()){
+				comparator.setComparisonLog(new ComparatorLogger(comparator.getClass().getName()));
+				addComparatorToLog();

```

Comment:
add comparator to parameters of addComparatorToLog() so the columns can be named after the comparator.
it is possible here to create a map that translates the attributes of the comparator's log to the attributes of the matching rule's log, so you can iterate over them later

Reply:
Done